src/operations/MessageOperation.cpp

```diff
@@ -59,9 +57,17 @@ MessageOperationFactory *MessageOperation::factory()
     return m_factory;
 }
 
-void MessageOperation::onMessageUpdate(const MessageUpdate &update)
+void MessageOperation::apply(const MessageUpdate &update)
 {
-    // TODO(fpohtmeh): use update result?
     m_message->applyUpdate(update);
-    // TODO(fpohtmeh): create json extras update here
+    emit updateMessage(update);
+
+    if (m_message->applyUpdate(update)) {
```

Comment:
Called `m_message->applyUpdate(update);` twice.
Can we reduce this calls to the one call?

Reply:
Sure. I pushed the fix